scripts/retrieve_cutouts.py

```diff
@@ -0,0 +1,97 @@
+# SPDX-FileCopyrightText:  Open Energy Transition gGmbH
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+
+import os
+import sys
+sys.path.append(os.path.abspath(os.path.join(__file__ ,"../../")))
+from pathlib import Path
+from google_drive_downloader import GoogleDriveDownloader as gdd
+import re
+from zipfile import ZipFile
+import yaml
+import logging
+import warnings
+warnings.filterwarnings("ignore")
+from scripts._helper import mock_snakemake, update_config_from_wildcards, create_logger, PYPSA_EARTH_DIR
+
+
+logger = create_logger(__name__)
+
+
+def download_and_unzip_gdrive(config, disable_progress=False):
+    """
+        Downloads and unzips USA cutouts to submodules/pypsa-earth/cutouts folder 
+    """
+    resource = config["category"]
+    file_path = os.path.join(PYPSA_EARTH_DIR, "tempfile.zip")
+    destination = os.path.join(PYPSA_EARTH_DIR, config["destination"])
+    url = config["urls"]["gdrive"]
+
+    # retrieve file_id from path
+    try:
+        # cut the part before the ending \view
+        partition_view = re.split(r"/view|\\view", str(url), 1)
+        if len(partition_view) < 2:
+            logger.error(
+                f'Resource {resource} cannot be downloaded: "\\view" not found in url {url}'
+            )
+            return False
+
+        # split url to get the file_id
+        code_split = re.split(r"\\|/", partition_view[0])
+
+        if len(code_split) < 2:
+            logger.error(
+                f'Resource {resource} cannot be downloaded: character "\\" not found in {partition_view[0]}'
+            )
+            return False
+
+        # get file id
+        file_id = code_split[-1]
+
+        # remove tempfile.zip if exists
+        Path(file_path).unlink(missing_ok=True)
```

Comment:
This piece of code removes tempfile, if it exists. If it does not exist, it will not throw error. Such piece of code is widely used in pypsa-earth `retrieve_databundle_light` script, so I thought it is a good practice to remove if exists.

Reply:
Then we can leave it as it is, ensuring best practices.